lib/sublime-block-comment.js

```diff
@@ -11,15 +11,31 @@ export default {
     const buffer = editor.getBuffer();
 
     buffer.transact(() => {
-      for (const selection of editor.getSelectionsOrderedByBufferPosition().reverse()) {
-        const range = selection.getBufferRange();
-        const textInRange = buffer.getTextInRange(range);
-
+      for (let i in editor.getSelectionsOrderedByBufferPosition().reverse()) {
+        const selection = editor.getSelectionsOrderedByBufferPosition().reverse()[i];
+        const langs = editor.scopeDescriptorForBufferPosition(selection.getBufferRange().start).getScopesArray();
         // TODO change comment syntax based on scope's grammar
-        const commentTokens = {
+        let commentTokens = {
           open:  '/*',
-          close: '*/',
+          close: '*/'
         };
+        for (let lang of langs) {
+          if (lang.indexOf('text.html') > -1) {
+            commentTokens = {
+              open:  '<!--',
+              close: '-->',
+            };
+          }
+          if ((lang.indexOf('source.js.embedded.html') > -1) ||
```

Comment:
So you think is better use somethink like this? (testing for embedded languages maybe in separated function)
```javascript
if (lang[0].includes('text.html')) {
          commentTokens = {
            open:  '<!--',
            close: '-->'
          };
        }
        if (lang[1].includes('source.js.embedded.html') ||
            lang[1].includes('source.css.embedded.html')) {
              commentTokens = {
                open:  '/*',
                close: '*/'
              };
            } 
```